uk.co.compucorp.civicrm.hrcore/templates/CRM/HRCore/Form/Search/SearchCriteria.tpl

```diff
@@ -0,0 +1,77 @@
+{if $context EQ 'smog'}
+  {capture assign=editTitle}{ts}Find Contacts within this Group{/ts}{/capture}
+{elseif $context EQ 'amtg' AND !$rows}
+  {capture assign=editTitle}{ts}Find Contacts to Add to this Group{/ts}{/capture}
+{else}
+  {capture assign=editTitle}{ts}Edit Search Criteria{/ts}{/capture}
+{/if}
+
+{strip}
+  <div class="crm-block crm-form-block crm-basic-criteria-form-block">
+    <div class="crm-accordion-wrapper crm-case_search-accordion {if $rows}collapsed{/if}">
+      <div class="crm-accordion-header crm-master-accordion-header">
+        {$editTitle}
+      </div>
+      <div class="crm-accordion-body">
+
+        <div class="crm-section" id="select-staff">
+          <div class="label">
+            {$form.select_staff.label}
+          </div>
+          <div class="content">
+            {$form.select_staff.html}
+          </div>
+          <div class="clear"></div>
+        </div>
+
+        <div class="contract-dates">
+          <div class="crm-section">
+            <div class="label">
+              {$form.contract_start_date.label}
+            </div>
+            <div class="content">
+              {$form.contract_start_date.html}
+            </div>
+          </div>
+
+          <div class="crm-section">
+            <div class="label">
+              {$form.contract_end_date.label}
+            </div>
+            <div class="content">
+              {$form.contract_end_date.html}
+            </div>
+          </div>
+        </div>
+
+        <script type="text/javascript">
+          {literal}
+          CRM.$(function($) {
+            $(document).ready(function() {
+              toogleContractDates();
+              initSelectStaffControls();
+            });
+
+            function initSelectStaffControls() {
+              $('#select-staff select').on('change', function(e) {
+                toogleContractDates();
+              });
+            }
+
+            function toogleContractDates() {
+              var select_staff_value = $('#select-staff select option:selected').val();
+              if (select_staff_value === 'choose_date') {
+                $('.contract-dates').show()
+              } else {
+                $('.contract-dates').hide();
+              }
+            }
+          });
+          {/literal}
```

Comment:
Civi already provides a date range filter that allows a specific range to be informed when a "Choose Date" option is selected. I think it just works out of the box, without need any kind of custom JS code like this. Could you please give more details about why you need this custom code?

You can have an example of this filter I'm talking about at the date fields of the Job Contract panel of in the Advanced Search

Reply:
Yes thats true, however, the field is built like so: https://github.com/compucorp/civihr/blob/e2ea5d1a0ddc70fbaeb066b05629c9eebf0eb84f/hrjobcontract/CRM/Hrjobcontract/BAO/Query.php#L426. If we add additional options like [All Staff, Current Staff], in the options array, Civi will throw an error when evaluating these values after submission because it will consider them to be relative dates and these options are not part of civi recognized relative dates processed internally by the `CRM_Utils_Date::relativeToAbsolute` function

Reply:
This is an important technical decision about a limitation from CiviCRM but it was documented neither in the technical details section nor in the code. Please add documentation to both to make sure all the devs will be able to understand the limitation and your technical solution